### PR TITLE
Deduplicate the ros1/ros2 code in the Action CI yaml

### PR DESCRIPTION
**Pull request by emersonknapp**
Was able to remove duplicated lines of code by utilizing the build matrix more.

### CODE REVIEW

File: .github/workflows/build_and_test_release_latest.yml [hunk 1]
```diff
@@ -60,25 +34,24 @@ jobs:
...
     # Needed for the CMakeLists.txt setup
-    - run: export ROS_VERSION=1
+    - run: export ROS_VERSION=${{ matrix.ros_version }}
     - uses: ros-tooling/action-ros-ci@0.0.17
       with:
         source-ros-binary-installation: ${{ matrix.ros_distro }}
         package-name: aws_common
-        extra-cmake-args: --cmake-target tests
-        # schedule only works for the master branch, so specify release-latest
+        # schedule runs against the default branch (master), so specify release-latest via repos file
```
Comment (dabonnie) [inline .github/workflows/build_and_test_release_latest.yml:42]: Why was `extra-cmake-args` removed?

Comment (emersonknapp) [reply .github/workflows/build_and_test_release_latest.yml:42]: Accident! I meant to work through making sure everything was passing before requesting review. It's been re-added

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Build & Test utils_common
```
Comment (dabonnie) [inline .github/workflows/build_and_test.yml:1]: As @thomas-moulard pointed out to me, you'll need to make sure the branch protection rules are up to date with any name changes to the actions.

File: .github/workflows/build_and_test_release_latest.yml [hunk 3]
```diff
@@ -1,54 +1,31 @@
-name: Build & Test utils_common release-latest
```
Comment (dabonnie) [inline .github/workflows/build_and_test_release_latest.yml:1]: To check you can add `pull_request` to this patch to verify your changes will work before merging. After verifying I would then remove.

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (emersonknapp) [reply .github/workflows/build_and_test.yml:1]: Yes, but I won't change it until after this PR is checked in

File: .github/workflows/build_and_test_release_latest.yml (hunk 3 above)
Comment (emersonknapp) [reply .github/workflows/build_and_test_release_latest.yml:1]: Done, passed. reverting now

**Pull request merged**